### PR TITLE
Document tx selection time options

### LINKED ISSUE #1489: Document block transaction selection max time options
These options were originally hidden, but should now be documented.
For more information, see: https://github.com/hyperledger/besu/pull/6423

### PR DESCRIPTION
**Pull request by alexandratran**
This PR documents `--block-txs-selection-max-time` and `--poa-block-txs-selection-max-time`. Fixes #1489.

Preview:
- https://besu-docs-u9dalbgo8-hyperledger.vercel.app/development/public-networks/reference/cli/options#block-txs-selection-max-time
- https://besu-docs-u9dalbgo8-hyperledger.vercel.app/development/private-networks/reference/cli/options#poa-block-txs-selection-max-time

### CODE REVIEW

**Review by fab-10 — APPROVED** [review]

File: docs/private-networks/reference/cli/options.md [hunk 1]
```diff
@@ -427,6 +427,54 @@ permissions-nodes-contract-version=2
...
+
+</Tabs>
+
+The maximum time that can be spent selecting transactions to be included in a block,
+as a percentage of the fixed block time of the network.
+The default is `75`, or 75%.
+
+:::note
+This option only applies to proof of authority networks.
+For proof of stake and proof of work networks, see
```
Comment (joaniefromtheblock) [inline docs/private-networks/reference/cli/options.md:474]:
```suggestion
For proof-of-stake and proof-of-work networks, see
```

File: docs/private-networks/reference/cli/options.md [hunk 2]
```diff
@@ -427,6 +427,54 @@ permissions-nodes-contract-version=2
...
+</TabItem>
+
+</Tabs>
+
+The maximum time that can be spent selecting transactions to be included in a block,
+as a percentage of the fixed block time of the network.
+The default is `75`, or 75%.
+
+:::note
+This option only applies to proof of authority networks.
```
Comment (joaniefromtheblock) [inline docs/private-networks/reference/cli/options.md:473]:
```suggestion
This option only applies to proof-of-authority networks.
```

File: docs/public-networks/reference/cli/options.md [hunk 3]
```diff
@@ -271,6 +271,53 @@ The singular `--banned-node-id` and plural `--banned-node-ids` are available and
...
+</TabItem>
+
+</Tabs>
+
+The maximum time, in milliseconds, that can be spent selecting transactions to be included in a block.
+This value must be less than or equal to the default, `5000`.
+
+:::note
+This option only applies to proof of stake and proof of work networks.
+For proof of authority networks, see
```
Comment (joaniefromtheblock) [inline docs/public-networks/reference/cli/options.md:317]:
```suggestion
For proof-of-authority networks, see
```

**Review by joaniefromtheblock — APPROVED** [review]
LGTM. Just a minor suggestion

We are somewhat inconsistent with how we format proof-of-stake and proof-of-work, but we link out to [these](https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/) docs that format it as proof-of stake.

File: docs/public-networks/reference/cli/options.md [hunk 4]
```diff
@@ -271,6 +271,53 @@ The singular `--banned-node-id` and plural `--banned-node-ids` are available and
...
+
+</TabItem>
+
+</Tabs>
+
+The maximum time, in milliseconds, that can be spent selecting transactions to be included in a block.
+This value must be less than or equal to the default, `5000`.
+
+:::note
+This option only applies to proof of stake and proof of work networks.
```
Comment (alexandratran) [inline docs/public-networks/reference/cli/options.md:316]:
```suggestion
This option only applies to proof-of-stake and proof-of-work networks.
```

**Pull request merged**